client/ethereum.go

```diff
@@ -25,6 +27,12 @@ var OneEth = big.NewFloat(1e18)
 type EthereumClient struct {
 	Client  *ethclient.Client
 	Network BlockchainNetwork
+
```

Comment:
Would you mind to add nonce handling there too? Or I can merge it from volume branch as an additional option later

Reply:
I'd merge it from the volume branch imo. I don't want to cherry-pick/reimplement and then have to deal with conflicts when you already have it working.

Reply:
Good, just asking if you ok that I'll add nonce handling to that code later.